cluster_experiments/experiment_analysis.py

```diff
@@ -111,6 +113,26 @@ def get_point_estimate(self, df: pd.DataFrame) -> float:
         self._data_checks(df=df)
         return self.analysis_point_estimate(df)
 
+    def pvalue_based_on_hypothesis(self, model_result) -> float:
```

Comment:
this only works for regression models, no? if it doesn't work for all kinds of analysis, I wouldn't put in here. I would use an external class or function that does this, and use composition instead of inheritance

Reply:
if we don't have a way to handle right-tailed or left-tailed for the t-test and paired t-test, we can always explicitely raise an error if left or right are passed to that test